File: 2018/python/day01.py

with open("./inputs/day01.txt") as f:
    data = f.read()

# Part 1

freq = 0

for line in data.split():
    if line.strip() == "":
        continue
    freq = freq + int(line)

print( "Solution 1: ", freq )

# Part 2

uniqueFreq = set()
should_loop_continue = True

freq = 0

while should_loop_continue:

    for line in data.split():
        uniqueFreq.add(freq)
        if line.strip() == "":
            continue
        freq = freq + int(line)

        if freq in uniqueFreq:
            should_loop_continue = False
            break

print( "Solution 2: ", freq )

File: 2018/nim/day01.nim
import strutils
import intsets

const data = readFile("./inputs/day01.txt")

# Part 1

var freq = 0

for line in data.splitLines():
    if line.strip() == "":
        continue
    freq = freq + parseInt(line)

echo "Solution 1: ", freq

# Part 2

var uniqueFreq : IntSet = initIntSet()
var should_loop_continue = true

freq = 0

while should_loop_continue:

    for line in data.splitLines():
        uniqueFreq.incl(freq)
        if line.strip() == "":
            continue
        freq = freq + parseInt(line)

        if freq in uniqueFreq:
            should_loop_continue = false
            break

echo "Solution 2: ", freq
